### PR TITLE
Fix 404 when setting node labels on node create

### LINKED ISSUE #27: Automatic setting of labels fails with 404
Creating nodes end up hitting 404 on server api while setting labels:
```
E, [2017-10-30T11:09:03.117351 #17123] ERROR -- CLI: Not found (/v1/nodes/prod/prod/strong-cloud-12) (Kontena::Errors::StandardError)
/usr/local/Cellar/kontena/1.4.0/libexec/gems/kontena-cli-1.4.0/lib/kontena/client.rb:537:in `handle_error_response'
/usr/local/Cellar/kontena/1.4.0/libexec/gems/kontena-cli-1.4.0/lib/kontena/client.rb:343:in `rescue in request'
/usr/local/Cellar/kontena/1.4.0/libexec/gems/kontena-cli-1.4.0/lib/kontena/client.rb:284:in `request'
/usr/local/Cellar/kontena/1.4.0/libexec/gems/kontena-cli-1.4.0/lib/kontena/client.rb:219:in `put'
/Users/foo/.kontena/gems/2.4.2/gems/kontena-plugin-upcloud-0.3.1/lib/kontena/machine/upcloud/node_provisioner.rb:114:in `set_labels'
/Users/foo/.kontena/gems/2.4.2/gems/kontena-plugin-upcloud-0.3.1/lib/kontena/machine/upcloud/node_provisioner.rb:91:in `block in run!'
/Users/foo/.kontena/gems/2.4.2/gems/kontena-plugin-upcloud-0.3.1/lib/kontena/machine/upcloud/node_provisioner.rb:36:in `times'
/Users/foo/.kontena/gems/2.4.2/gems/kontena-plugin-upcloud-0.3.1/lib/kontena/machine/upcloud/node_provisioner.rb:36:in `run!'
/Users/foo/.kontena/gems/2.4.2/gems/kontena-plugin-upcloud-0.3.1/lib/kontena/plugin/upcloud/nodes/create_command.rb:23:in `execute'
/usr/local/Cellar/kontena/1.4.0/libexec/gems/kontena-cli-1.4.0/lib/kontena/command.rb:213:in `run'
/usr/local/Cellar/kontena/1.4.0/libexec/gems/clamp-1.1.2/lib/clamp/subcommand/execution.rb:11:in `execute'
/usr/local/Cellar/kontena/1.4.0/libexec/gems/kontena-cli-1.4.0/lib/kontena/command.rb:213:in `run'
/usr/local/Cellar/kontena/1.4.0/libexec/gems/clamp-1.1.2/lib/clamp/subcommand/execution.rb:11:in `execute'
/usr/local/Cellar/kontena/1.4.0/libexec/gems/kontena-cli-1.4.0/lib/kontena/command.rb:213:in `run'
/usr/local/Cellar/kontena/1.4.0/libexec/gems/clamp-1.1.2/lib/clamp/subcommand/execution.rb:11:in `execute'
/usr/local/Cellar/kontena/1.4.0/libexec/gems/kontena-cli-1.4.0/lib/kontena/command.rb:213:in `run'
/usr/local/Cellar/kontena/1.4.0/libexec/gems/clamp-1.1.2/lib/clamp/command.rb:132:in `run'
/usr/local/Cellar/kontena/1.4.0/libexec/gems/kontena-cli-1.4.0/bin/kontena:18:in `<top (required)>'
/usr/local/Cellar/kontena/1.4.0/libexec/bin/kontena:23:in `load'
/usr/local/Cellar/kontena/1.4.0/libexec/bin/kontena:23:in `<main>'
```

https://github.com/kontena/kontena-plugin-upcloud/blob/master/lib/kontena/machine/upcloud/node_provisioner.rb#L114

So for a grid named `prod`, it ends up "duplicating" the grid name in the node identifier in the URL.

### PR DESCRIPTION
**Pull request by kke**
Fixes #27

### CODE REVIEW

**Comment by kke** [comment]
image
image

Implementation = copypaste from aws plugin

File: lib/kontena/machine/upcloud/node_provisioner.rb [hunk 1]
```diff
@@ -109,9 +109,12 @@ def erb(template, vars)
...
         end
 
+        # @param [Hash] node
+        # @param [Array<String>] labels
         def set_labels(node, labels)
-          data = {labels: labels}
-          api_client.put("nodes/#{node['grid']['id']}/#{node['id']}", data)
+          data = {}
+          data[:labels] = labels
+          api_client.put("nodes/#{node['id']}", data, {}, {'Kontena-Grid-Token' => node['grid']['token']})
```
Comment (SpComb) [inline lib/kontena/machine/upcloud/node_provisioner.rb]: This should not need the `Kontena-Grid-Token` header if the `api_client` is using the normal oauth bearer token.

**Review by SpComb — COMMENTED** [review]
Otherwise LGTM apart from the unnecessary (?) `Kontena-Grid-Token` header.

File: lib/kontena/machine/upcloud/node_provisioner.rb (hunk 1 above)
Comment (jakolehm) [reply lib/kontena/machine/upcloud/node_provisioner.rb]: Yeah, this header does not look right?

Comment (kke) [reply lib/kontena/machine/upcloud/node_provisioner.rb]: Strange that it has been there for two years unchanged in AWS plugin: https://github.com/kontena/kontena-plugin-aws/blame/master/lib/kontena/machine/aws/node_provisioner.rb#L201

Comment (kke) [reply lib/kontena/machine/upcloud/node_provisioner.rb]: Seems to work without.

**Review by jakolehm — APPROVED** [review]

**Pull request merged**